### PR TITLE
MYNEWT-583; newt - allow use of specific tag/repository revision.

### PR DESCRIPTION
**Pull request by mkiiskila**

### CODE REVIEW

File: newt/repo/repo.go [hunk 1]
```diff
@@ -157,6 +173,32 @@ func NewRepoDependency(rname string, verstr string) (*RepoDependency, error) {
...
 
+func pickVersion(repo *Repo, versions []*Version) ([]*Version, error) {
+	fmt.Printf("Dependency list for %s contains a specific commit tag, " +
+		"so normal version number/stability comparison cannot be done.\n",
+		repo.Name())
+	fmt.Printf("If the following list does not contain the requirement to use, "+
+		"then modify your project.yml so that it does.\n")
+	for {
+		for i, vers := range versions {
+			fmt.Printf(" %d) %s\n", i, vers)
```
Comment (ccollins476ad) [inline newt/repo/repo.go]: Should "vers" be converted to a string here?

Comment (mkiiskila) [inline newt/repo/repo.go]: I believe it calls the String() method for it automagically, presumably due to %s on the format string.
However, I can add the explicit call to make it more clear what's going on.

**Comment by mkiiskila** [comment]
Updated pull request.

**Comment by ccollins476ad** [comment]
Thanks, Marko.  I don't think String() gets called automatically.  At least, it didn't when I tried it in the Go playground.  I did that test a while ago and no longer have the exact code I tested, though.

**Comment by ccollins476ad** [comment]
Just an FYI- here is a playground program which I think reproduces this case: https://play.golang.org/p/UP8i8vo1oZ

```
package main

import (
	"fmt"
)

type S struct {
	a int
}

func (s *S) String() string {
	return fmt.Sprintf("%d", s.a)
}

func main() {
	s := S{5}
	fmt.Printf("s=%s\n", s)
}
```

Output:
```
s={%!s(int=5)}

Program exited.
```

Maybe there is something different in this example?

**Comment by ccollins476ad** [comment]
Btw, I figured out why my playground example wasn't working.  The String() function operated on a `*S`, but I was passing a plain `S` to Printf.  I updated the example code and now it does work (https://play.golang.org/p/4kMNJ8NdEB):

```
package main

import (
	"fmt"
)

type S struct {
	a int
}

func (s *S) String() string {
	return fmt.Sprintf("%d", s.a)
}

func main() {
	s := &S{5}
	fmt.Printf("s=%s\n", s)
}
```

```
s=5

Program exited.
```

**Pull request closed without merging**